File: src/main/java/com/magichamster/grocerysamurai/model/GroceryListItem.java
package com.magichamster.grocerysamurai.model;

import javax.persistence.*;

/**
 * Table of items on a grocery list Created by Rick on 4/14/17.
 */
@Entity
@Table(name = "grocery_list_item")
public class GroceryListItem extends Identity {
	@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
	@JoinColumn(name = "grocery_list_id", nullable = false)
	private GroceryList groceryList;

	@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
	@JoinColumn(name = "item_id", nullable = false)
	private Item item;

	@Column(name = "quantity")
	private int quantity;

	@Column(name = "package_type")
	private String packageType;

	@Column(name = "weight")
	private int weight;

	public int getWeight() {
		return weight;
	}

	public void setWeight(int weight) {
		this.weight = weight;
	}

	public GroceryListItem() {
	}

	public GroceryList getGroceryList() {
		return groceryList;
	}

	public void setGroceryList(GroceryList groceryList) {
		this.groceryList = groceryList;
		groceryList.getGroceryListItems().add(this);
	}

	public Item getItem() {
		return item;
	}

	public void setItem(Item item) {
		this.item = item;
		item.getGroceryListItems().add(this);
	}

	public int getQuantity() {
		return quantity;
	}

	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}

	public String getPackageType() {
		return packageType;
	}

	public void setPackageType(String packageType) {
		this.packageType = packageType;
	}
}

File: MagicHamster.GrocerySamurai/MagicHamster.GrocerySamurai.Model/Entities/GroceryListItem.cs
namespace MagicHamster.GrocerySamurai.Model.Entities
{
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using Common;
    using JetBrains.Annotations;

    [Table("grocery_list_item")]
    public class GroceryListItem : Entity
    {
        [Column("grocery_list_id")]
        [UsedImplicitly]
        public int GroceryListId { get; set; }

        [Column("item_id")]
        [UsedImplicitly]
        public int ItemId { get; set; }

        [Column("quantity")]
        [UsedImplicitly]
        public int Quantity { get; set; }

        [Column("package_type")]
        [MaxLength(30)]
        [Display(Name = "Package Type")]
        [UsedImplicitly]
        public string PackageType { get; set; }

        [Column("weight")]
        [UsedImplicitly]
        public int Weight { get; set; }

        [ForeignKey("GroceryListId")]
        [UsedImplicitly]
        public GroceryList GroceryList { get; set; }

        [ForeignKey("ItemId")]
        [UsedImplicitly]
        public Item Item { get; set; }
    }
}
